io-engine/tests/nexus_rebuild_partial_loop.rs

```diff
@@ -0,0 +1,255 @@
+        .name("cargo-test")
+        .network("10.1.0.0/16")
+        .unwrap()
+        .add_container_bin(
+            "ms_0",
+            Binary::from_dbg("io-engine")
+                .with_args(vec!["-l", "1,2", "-Fcolor,compact,nodate,host"])
+                .with_bind("/tmp", "/host/tmp"),
+        )
+        .add_container_bin(
+            "ms_1",
+            Binary::from_dbg("io-engine")
+                .with_args(vec!["-l", "3,4", "-Fcolor,compact,nodate,host"])
+                .with_bind("/tmp", "/host/tmp"),
+        )
+        .add_container_bin(
+            "ms_2",
+            Binary::from_dbg("io-engine")
+                .with_args(vec!["-l", "5,6", "-Fcolor,compact,nodate,host"])
+                .with_bind("/tmp", "/host/tmp"),
+        )
+        .with_clean(true)
+        .build()
+        .await
+        .unwrap();
+
+    let conn = GrpcConnect::new(&test);
+
+    // Nexus node handle.
+    let ms_nex = conn.grpc_handle_shared("ms_2").await.unwrap();
+
+    // Create node objects.
+    let mut nodes = Vec::new();
+
+    for (idx, name) in ["ms_0", "ms_1", "ms_2"].iter().enumerate() {
+        let ms = conn.grpc_handle_shared(name).await.unwrap();
+
+        let file_name = format!("{DISK_NAME}_{idx}");
+
+        // Create pool data file.
+        let data_file_path = format!("/tmp/{file_name}");
+        common::delete_file(&[data_file_path.clone()]);
+        common::truncate_file_bytes(&data_file_path, DISK_SIZE);
+
+        let bdev_name = format!("aio:///host/tmp/{file_name}?blk_size=512");
+
+        let mut pool = PoolBuilder::new(ms.clone())
+            .with_name(&format!("p{idx}"))
+            .with_new_uuid()
+            .with_bdev(&bdev_name);
+
+        pool.create().await.unwrap();
+
+        let mut repl = ReplicaBuilder::new(ms.clone())
+            .with_pool(&pool)
+            .with_name(&format!("v0r{r}", r = idx))
+            .with_new_uuid()
+            .with_size_mb(REPL_SIZE)
+            .with_thin(false);
+
+        repl.create().await.unwrap();
+        repl.share().await.unwrap();
+
+        nodes.push(Node {
+            name: name.to_string(),
+            ms,
+            data_file_path,
+            pool,
+            repl,
+        });
+    }
+
+    // Create the nexus.
+    let mut nex = NexusBuilder::new(ms_nex.clone())
+        .with_name(&format!("v0"))
+        .with_new_uuid()
+        .with_size_mb(NEXUS_SIZE)
+        .with_replica(&nodes[0].repl)
+        .with_replica(&nodes[1].repl)
+        .with_replica(&nodes[2].repl);
+
+    nex.create().await.unwrap();
+    nex.publish().await.unwrap();
+
+    // Notifier.
+    let (sender, mut recv) = oneshot::channel();
+
+    let j0 = tokio::spawn({
+        let nvmf = nex.nvmf_location();
+        async move {
+            println!("[A]");
+            let (_cg, path) = nvmf.open().unwrap();
+            println!("[A] nvme path: {path:?}");
+
+            let fio = FioBuilder::new()
+                .with_fio_binary(
```

Comment:
Why do we have the need to provide fio binary explicitly, to rule out any fio bugs in default 3.33 ?

Reply:
Yes, Fio builder API for providing fio binary explicitly is needed for develpoment / debug purposes.
But this one is, again, I forgot to clean up.